### ISSUE TITLE
Add Simple LED strip Control?

### ISSUE THREAD

**Issue by GeorgeIoak** [penfold42/ESPixelBoard#46]
I was wondering if there was an "easy" way to add support for a new type of LEDs. I made a short video of the LED string and it's different lighting effects. I made some notes in the video description of how this strip is working.

It seems relatively easy to make these effects (or others) with code and directly connecting the LED string to an ESP8266 running the ESPixel firmware. I'd like to integrate these into my Xlights show. I just don't know how to take e1.31 and convert that to something that could drive the LEDs. Driving the LEDs is just simple PWM and switching the polarity using an H-Bridge. I know that this strip can't perform most of the effects but it would be nice to just include them with the other lights and have them do something.

**Comment by penfold42** [comment]
What’s the timing like ?
Have you documented the waveforms ?
Does it need to go negative/reverse polarity?

**Comment by GeorgeIoak** [comment]
I can measure the timing but it's not too critical. Yes, you reverse the polarity but they probably use a H-Bridge for that. There's a [WEMOS Mini Motor Shield](https://wiki.wemos.cc/products:d1_mini_shields:motor_shield) which I ordered that should work. Since the TB6612FNG chip on the shield supports 2 motors you can connect 2 strings instead of 2 motors.

**Comment by GeorgeIoak** [comment]
I took [some scope plots](https://imgur.com/a/z08RPRK) of the simple switching between the 2 series (pink/blue and yellow/green). I need to look at the circuit board but I think they're controlling the current by the PWM that's used during the on time for the LEDs,

**Comment by GeorgeIoak** [comment]
Can you point me in the right direction to add a new class for these? I'm thinking initially to just be able to accept data from xlights and then rather than trying to decode what to do just hard code in 1 mode for now, like breathing LEDs.

**Comment by GeorgeIoak** [comment]
Yes, there are only 2 wires on the strip with the LEDs connected directly across the 2 wires. I need to take the board out of the box and take a closer look but I don't think the board has any current limiting so they use PWM extensively to limit the current to the LEDs, or at least that's my guess since I never saw a flat line on the scope.

Directly sending a PWM would light up 2/4 of the LEDs so we need some hook to tell the H-Bridge to switch direction. The Wemos Motor Shield has a STM32 MCU on it which is used I guess to minimize the work load of the ESP. It seems that the connection between the shield and the ESP is done with I2C. [Wemos wrote a library](https://github.com/wemos/WEMOS_Motor_Shield_Arduino_Library) so there's very little code to be written to drive a motor or LED in this case.
